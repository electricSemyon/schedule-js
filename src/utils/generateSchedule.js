const generateSchedule = (items) => {
  const currentShedule = [];

  for(let i = 0; i < 7; i++) {
    currentShedule[i] = {
      body: items[Math.floor(Math.random()*items.length)].body,
      id: i
    };
  }

  return currentShedule;
}

export default generateSchedule;
