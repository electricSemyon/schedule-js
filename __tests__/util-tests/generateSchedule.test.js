import generateSchedule from '../../src/utils/generateSchedule.js';
import itemsJSON from '../fixtures/scheduleItems.json';

test('generates schedule, based on array of items', () => {
  const schedule = generateSchedule(itemsJSON.items);

  expect(schedule).toMatchSnapshot();
});
