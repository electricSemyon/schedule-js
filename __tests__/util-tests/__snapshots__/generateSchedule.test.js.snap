exports[`test generates schedule, based on array of items 1`] = `
Array [
  Object {
    "body": "bumming",
    "id": 0,
  },
  Object {
    "body": "bumming",
    "id": 1,
  },
  Object {
    "body": "bumming",
    "id": 2,
  },
  Object {
    "body": "bumming",
    "id": 3,
  },
  Object {
    "body": "bumming",
    "id": 4,
  },
  Object {
    "body": "bumming",
    "id": 5,
  },
  Object {
    "body": "bumming",
    "id": 6,
  },
]
`;
